src/App.js

```diff
@@ -1,67 +1,71 @@
 
+let IMAGE_HEIGHT = 200;
+
 class Image extends React.Component {
     constructor(props) {
         super(props);
-        this.addCat = this.addCat.bind(this);
-        this.handleScroll = this.handleScroll.bind(this);
-        this.imageHeigh = "200";
         this.state = {
-            isFirstTime: true,
+            isFetching: false,
             error: null,
             isLoaded: false,
             images: [],
             catsElements: [],
             makeTimeout: false
         };
-        window.addEventListener("scroll", this.handleScroll);
+
     }
 
     handleScroll = (e) => {
-        const bottom = e.target.scrollingElement.scrollHeight - this.imageHeigh - e.target.scrollingElement.scrollTop <= e.target.scrollingElement.clientHeight + 1;
+        const bottom = e.target.scrollingElement.scrollHeight - IMAGE_HEIGHT - e.target.scrollingElement.scrollTop <= e.target.scrollingElement.clientHeight + 1;
         if (bottom) {
             this.addCat();
         }
     }
 
+    componentDidMount() {
+        this.addCat();
+        window.addEventListener("scroll", this.handleScroll);
+    }
 
-    addCat() {
-        console.log("addCat")
-        fetch("https://api.thecatapi.com/v1/images/search?limit=10")
-            .then((response) => {
-                return response.json();
-            })
-            .then((arrayOfData) => {
-                const buffArray = [];
-                arrayOfData.map((data) => {
-                    console.log(data);
-                    buffArray.push(
-                        <div>
-                            <img src={data.url} width="200" height={this.imageHeigh}/>
-                        </div>
-                    );
-                });
-                this.setState({
-                    images: this.state.images.concat(buffArray)
-                });
-            });
+    componentWillUnmount() {
+        window.removeEventListener("scroll", this.handleScroll);
     }
 
 
-    render() {
-        const style = {
-            justifyContent: 'center',
-            alignItems: 'center',
-            position: 'absolute', // Найдите
-        }
-        if (this.state.isFirstTime) {
-            this.addCat()
+    addCat = () => {
+        if (!this.state.isFetching) {
+            console.log("addCat")
             this.setState({
-                isFirstTime: false
+                isFetching: true
             });
+            fetch("https://api.thecatapi.com/v1/images/search?limit=10")
+                .then((response) => {
+                    return response.json();
+                })
+                .then((arrayOfData) => {
+                    const buffArray = [];
+                    arrayOfData.forEach((imageData) => {
+                        console.log(imageData);
+                        buffArray.push(imageData);
+                    });
+                    this.setState({
+                        images: this.state.images.concat(buffArray),
```

Comment:
If you need to read the state when updating it (like reading `this.state.images` here), then It's recommended to pass a function (instead of an object) to the `this.setState` function. For example like this:
```js
                    this.setState(prevState => ({
                        images: prevState.images.concat(buffArray),
                        isFetching: false
                    }));
```